interfaces/builtin/online_accounts_service.go

```diff
@@ -0,0 +1,186 @@
+# Description: Allow operating as the Online Accounts service. Reserved because
+# this gives privileged access to the system.
+
+# DBus accesses
+#include <abstractions/dbus-session-strict>
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member={RequestName,ReleaseName,GetConnectionCredentials}
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+	bus=session
+	name="com.ubuntu.OnlineAccounts.Manager",
+`)
+
+var onlineAccountsServiceConnectedSlotAppArmor = []byte(`
+# Allow service to interact with connected clients
+dbus (receive, send)
+	bus=session
+	path=/com/ubuntu/OnlineAccounts{,/**}
+	interface=com.ubuntu.OnlineAccounts.Manager
+	peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var onlineAccountsServiceConnectedPlugAppArmor = []byte(`
+# Description: Allow using Online Accounts service. Common because the access
+# to user data is actually mediated by the Online Accounts service itself.
+# Usage: common
+
+#include <abstractions/dbus-session-strict>
+
+# Online Accounts v2 API
+dbus (receive, send)
+    bus=session
+    interface=com.ubuntu.OnlineAccounts.Manager
+    peer=(label=###SLOT_SECURITY_TAGS###),
+`)
+
+var onlineAccountsServicePermanentSlotSecComp = []byte(`
+# dbus
+accept
+accept4
+bind
+listen
+recv
+recvfrom
+recvmmsg
+recvmsg
+send
+sendmmsg
+sendmsg
+sendto
+shutdown
+`)
+
+var onlineAccountsServiceConnectedPlugSecComp = []byte(`
+# dbus
+recv
+recvmsg
+send
+sendto
+sendmsg
+`)
+
+var onlineAccountsServicePermanentSlotDBus = []byte(`
+<policy user="default">
+    <allow own="com.ubuntu.OnlineAccounts.Manager"/>
+    <allow send_destination="com.ubuntu.OnlineAccounts.Manager"/>
+    <allow send_interface="com.ubuntu.OnlineAccounts.Manager"/>
+</policy>
+`)
+
+var onlineAccountsServiceConnectedPlugDBus = []byte(`
+<policy context="default">
+    <deny own="com.ubuntu.OnlineAccounts.Manager"/>
+    <allow send_destination="com.ubuntu.OnlineAccounts.Manager"/>
+    <allow send_interface="com.ubuntu.OnlineAccounts.Manager"/>
+</policy>
+`)
+
+type OnlineAccountsServiceInterface struct{}
+
+func (iface *OnlineAccountsServiceInterface) Name() string {
+	return "online-accounts-service"
+}
+
+func (iface *OnlineAccountsServiceInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityDBus, interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityUDev, interfaces.SecurityMount:
+		return nil, nil
+	default:
+		return nil, nil
+	}
+}
+
+func (iface *OnlineAccountsServiceInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return onlineAccountsServiceConnectedPlugAppArmor, nil
```

Comment:
You need to do the substitutions for ###SLOT_SECURITY_TAGS### in onlineAccountsServiceConnectedPlugAppArmor here. See the mir interface for and example.